tests/test_query.py

```diff
@@ -361,7 +366,7 @@ def test_should_fetch_fields_of_prefetched_field():
     qs = Item.objects.filter(name='foo')
     items = gql_optimizer.query(qs, info)
     optimized_items = qs.prefetch_related(
-        Prefetch('children', queryset=Item.objects.only('id')),
+        Prefetch('children', queryset=Item.objects.only('id', 'parent__id')),
```

Comment:
The addition of `parent__id` to the sql query is not necessary to have a performant query in this case. This line was weird for me at first, but then I realized what this PR was fixing and I understood that there's no easy way to avoid adding this field when optimizing the query. So I think this PR is a good compromise. We can leave the `parent__id` so that the test pass.